packages/opentelemetry-core/src/common/version.ts

```diff
@@ -0,0 +1,18 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// this is autogenerated file, see scripts/version.js
```

Comment:
I love this solution :+1: great idea!!!

Reply:
You could still optionally make it a `const enum`, but that's a small iteration on this

Reply:
enum is a set of values but here is just one variable only. The things which Daniel suggested was to have one file for constants. So far we just need version, if we ever need more constants, then I would still keep autogenerated alone but create a `constants` and simply require the version there. But that would be something to implement in separate PR if we ever need such thing right ?

Reply:
I guess I views the `Constants` enum as somewhat of an example. I would think there would be various constant enums for different purposes.

For example, in this case, could we do something like this

```
// This is an auto-generated file, see ...
// This uses `const enum` so that the TypeScript compiler will inline the value.
export const enum Version {
    VALUE = '0.0.3',
}
```

Then in the code you would do `core.Version.VALUE`. It is a little bit less clean than just doing `core.VERSION`, but the benefit is that it inlines the value at compile time. I believe the eventual go in https://github.com/open-telemetry/opentelemetry-js/issues/538 is to make all enums in `core` be `const` so that the `core` package can just compile down to nothing. Doing an `export const` would give it some residual value that wouldn't compile away I believe.

cc/ @dyladan

Reply:
I think you are mixing 2 things here. The thing raised in #538 is about changing currently used enum just by adding `const` to them. But here I'm generating the version only and this is not any enum, - why it even would be? And once more if we ever need more of such constants then we should simply change it then, but for now, only version is needed, nothing more.

Reply:
@draffensperger I don't believe core will ever compile to nothing as it has to have at the very least the `initGlobalTracer` function.

Reply:
@dyladan I think given the fact that `core` will never compile down to nothing, then @obecny I agree that the extra indirection of `const enum` doesn't make sense here. I would support just doing `export const VERSION = '...'`